.github/workflows/pull-request.yaml

```diff
@@ -129,6 +129,35 @@ jobs:
         rm -rf /tmp/.buildx-cache
         mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  test-examples:
+    needs:  [build-docker]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        example_dir: [tracetest-jaeger, tracetest-opensearch, tracetest-tempo]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '^1.18'
+      - name: Build example
+        run: |
+          cd examples/${{ matrix.example_dir }}
+          sed -i 's/kubeshop\/tracetest:.*/kubeshop\/tracetest:pr-${{ github.event.pull_request.number }}/g' docker-compose.yml
+          docker-compose up -d
+      - name: Build CLI
+        run: |
+          cd cli
+          go build -o ./tracetest main.go
+          echo "http://localhost:8080" | ./tracetest configure -g
+      - name: Run example test
+        run: |
+          timeout 60 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:8080)" != "200" ]]; do sleep 5; done' || false
+          ./cli/tracetest test run -d examples/${{ matrix.example_dir }}/tests/list-tests.yaml --wait-for-result
```

Comment:
is it possible to run all the dogfooding test for each example? that way we enforce full compatibility for all backends

Reply:
We can but I rather work on that after we get the chaining tests right. I don't feel the bash solution is very stable right now and this might make our pipeline more brittle

Reply:
Since the dogfood step runs isolated from the frontend tests, I didn't see it randomly failing anymore, so I'd feel confident in doing that

Reply:
to make it work well, I think we would still use docker-compose and use it as both "integration" and "target"  instances. Is that right? So we don't have to maintain opensearch and tempo in our cluster.

Reply:
In order for that to work, we will have to change all our examples to make them use the pokemon api. I'd do that but in another PR.

Reply:
@schoren is that ok? can this be merged?